### DISCUSSION TITLE
asyncpg INSERT error

### DISCUSSION THREAD

**Discussion by robd003** [crate/crate#12269] [Q&A] [answered]
Has anyone seen this before `asyncpg.exceptions.InternalServerError: blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];`

When looking at the admin console for the table it's in good health with all shards available and nothing to indicate that it's "read-only"

Any ideas on how to debug this @proddata ?

**Comment by robd003** [comment, marked as answer]
Turns out this was a side-effect when we did performance testing on this cluster and hit the high water mark on the disk.

I fixed it by running `ALTER TABLE tablename SET ("blocks.read_only_allow_delete" = FALSE);` on every single table which was annoying...

**Reply by proddata** [reply]
There is a PR already merged that would remove that flag again, when the disk has free space again.
~~It will be back ported as well.~~

~~https://github.com/crate/crate/pull/12231~~

**Reply by mfussenegger** [reply]
> There is a PR already merged that would remove that flag again, when the disk has free space again.
It will be back ported as well.

I think you may have mixed that up. The PR to automatically remove that flag is https://github.com/crate/crate/pull/12214 and it won't be backported.
